src/main/java/com/iota/iri/network/pipeline/SolidifyStage.java

```diff
@@ -0,0 +1,108 @@
+ * passed from the {@link ReceivedStage} it will be placed into this stage to have the {@link TransactionSolidifier}
+ * check the solidity of the transaction. If the transaction is found to be solid, it will be passed forward to the
+ * {@link BroadcastStage}. If it is found to be unsolid, it is put through the solidity check so that missing reference
+ * transactions get requested. If the transaction is unsolid, a random solid tip is broadcast instead to keep the
+ * requests transmitting to neighbors.
+ */
+public class SolidifyStage implements Stage {
+    private static final Logger log = LoggerFactory.getLogger(SolidifyStage.class);
+
+    private TransactionSolidifier txSolidifier;
+    private TransactionValidator txValidator;
+    private TipsViewModel tipsViewModel;
+    private Tangle tangle;
+    private TransactionViewModel tip;
+
+    /**
+     * Constructor for the {@link SolidifyStage}.
+     *
+     * @param txSolidifier      Transaction solidifier implementation for updating transaction's solidity status
+     * @param txValidator       Transaction validator implementation for determining the validity of a transaction
+     * @param tipsViewModel     Used for broadcasting random solid tips if the subject transaction is unsolid
+     * @param tangle            A reference to the nodes DB
+     */
+    public SolidifyStage(TransactionSolidifier txSolidifier, TransactionValidator txValidator,
+                         TipsViewModel tipsViewModel, Tangle tangle){
+        this.txSolidifier = txSolidifier;
+        this.txValidator = txValidator;
+        this.tipsViewModel = tipsViewModel;
+        this.tangle = tangle;
+    }
+
+    /**
+     * Processes the payload of the {@link ProcessingContext} as a {@link SolidifyPayload}. First the transaction will
+     * be checked for solidity and validity. If the transaction is already solid or can be set solid quickly by the
+     * transaction validator, the transaction is passed to the {@link BroadcastStage}. If not, the transaction is
+     * added to the solidification queue, and a random solid tip is pulled form the {@link TipsViewModel} to be
+     * broadcast instead.
+     *
+     * @param ctx       The context to process
+     * @return          The output context, in most cases a {@link BroadcastPayload}.
+     */
+    @Override
+    public ProcessingContext process(ProcessingContext ctx){
+        try {
+            SolidifyPayload payload = (SolidifyPayload) ctx.getPayload();
+            TransactionViewModel tvm = payload.getTransaction();
+
+            if (tvm.isSolid() || txValidator.quickSetSolid(tvm)) {
+                ctx.setNextStage(TransactionProcessingPipeline.Stage.BROADCAST);
+                ctx.setPayload(new BroadcastPayload(payload.getOriginNeighbor(), payload.getTransaction()));
+                return ctx;
+            }
+
+            txSolidifier.addToSolidificationQueue(tvm.getHash());
```

Comment:
Totally missed...
Delete this line :-P

Explanation:
What you are basically doing here is reviving the old tip solidifier...
We decided that we don't need to actively request transactions that are not needed to solidify milestones.
This is because: 
a. It is most likely that most txs will get quickly solid
b. If not due to receiving txs out of order then they should get solid by `propagateSolidTransactions` (this is optional and could be left out, maybe)
c. Else they may either become solid by milestone or never become solid

Basically, we assume that a&b will be enough most of the time, and if not a milestone will fix the situation.
If a milestone never comes then it doesn't matter because they are not in a consensus state.

The problems with actively requesting transactions for solidifying are:
1. We are wasting bandwidth to request txs that will not be approved
2. A malicious actor can slow us down by sending txs that approve imaginary txs